### PR TITLE
boards/stm32f1/f3: rework clock initialization and configuration

### PR DESCRIPTION
**Pull request by aabadie**
### Contribution description

This PR is merging STM32 F1 and F3 clock initialization with the F0 version that was split from other Fx families in #14921. It also rework the clock configuration to make it easy to select the clock source and configure the PLL prescalers.

F0/F1/F3 have a similar PLL configuration scheme so it makes more sense to initialize them in the same file. It also simplifies the migration toward a consistent clock configuration/initialization across STM32 cpu families.

This PR also add Kconfig code to configure the clock. The Kconfig file is shared between f0, f1 and f3 families.

There's also a last commit that renames the `stmclk_fx.c` to `stmclk_f2f4f7.c`. This new file is the last refactoring part that remains.

### Testing procedure

- A green Murdock
- Test any f1 and f3 based board, play with clock configuration (using CFLAGS and Kconfig) and verify that everything is functional
- Testing the f0 is also worth since some common parts are touched by this PR.

### Issues/PRs references

This PR is based on #14921 ~and #14892~ 
Related to #14975

### CODE REVIEW

File: cpu/stm32/stmclk/stmclk_f2f4f7.c [hunk 1]
```diff
@@ -38,8 +38,6 @@
  * @name    PLL configuration
```
Comment (fjmolinas) [inline cpu/stm32/stmclk/stmclk_f2f4f7.c:38]: unrelated

**Comment by fjmolinas** [comment]
Are there real conns of keeping the configuration separated? as more and more are mixed I tend to find the files hard to read..

**Comment by aabadie** [comment]
> Are there real conns of keeping the configuration separated?

Do you mean f0 against f1/f3 clock configuration headers ?

**Comment by fjmolinas** [comment]
> > Are there real conns of keeping the configuration separated?
> 
> Do you mean f0 against f1/f3 clock configuration headers ?

Yes, I'm just saying this because I can have trouble following files witch chains of `ifdef`s

**Comment by fjmolinas** [comment]
> > > Are there real conns of keeping the configuration separated?
> > 
> > 
> > Do you mean f0 against f1/f3 clock configuration headers ?
> 
> Yes, I'm just saying this because I can have trouble following files witch chains of `ifdef`s

Just realized that before there was a single `stmclk_fx`, so its to late to bring this up :)

**Comment by fjmolinas** [comment]
Ran some timing tests on one of each boards, https://ci.inria.fr/ci-riot-tribe/job/build-pipeline.jk/185/artifact/

**Comment by aabadie** [comment]
I can see what I can I do but I think the main reason is the maximum core clock: it's 48MHz on f0 and 72MHz on f1/f3.

**Comment by fjmolinas** [comment]
> Ran some timing tests on one of each `CPU`, https://ci.inria.fr/ci-riot-tribe/job/build-pipeline.jk/185/artifact/

Otherwise code looks good, I don't have the hardware I need to paly with the configuartion ATM, so I'll resume next week,

File: cpu/stm32/stmclk/stmclk_f2f4f7.c (hunk 1 above)
Comment (aabadie) [reply cpu/stm32/stmclk/stmclk_f2f4f7.c:38]: I think it is: this file is adapted with the removal of code related to f1 and f3. And the name is changed to reflect that now it only contains code related to f2f4f7. Keeping fx suffix makes no sense since it doesn't apply anymore to all Fx families.

Comment (fjmolinas) [reply cpu/stm32/stmclk/stmclk_f2f4f7.c:38]: Ah ok

**Comment by fjmolinas** [comment]
There seems to be an issue with stm32f0, see https://ci.inria.fr/ci-riot-tribe/job/build-pipeline.jk/188/artifact/ for details. Its also present in master so it must have sliped in from previous PR's, any ideas?

```
Slept for 98777 us (expected: 98765 us) Offset: 12 us
Slept for 75012 us (expected: 75000 us) Offset: 12 us
Test ran for 1649701 us
Host timer measured 3299020 us (client measured 1649701 us)
```

https://ci.inria.fr/ci-riot-tribe/job/build-pipeline.jk/188/artifact/vim

**Comment by fjmolinas** [comment]
There seems to be now issue with the f1 and f3.

**Comment by fjmolinas** [comment]
Offline @aabadie found the issue with f0, it comes from a previous PR so will be fixed on another PR.

**Review by fjmolinas — APPROVED** [review]
ACK

**Comment by aabadie** [comment]
Awesome! Thanks for reviewing @fjmolinas !
